makefile.shade

```diff
@@ -628,85 +541,97 @@ var RC_FILES = '${FindAllFiles("Resource.rc", BOOTSTRAPPER_EXE_NAME, BOOTSTRAPPE
+                            Log.Info("Copying " + file + " to " + targetFilePath);
+                            File.Copy(file, targetFilePath, true);
+                        }
                     }
                 }
-                else
+                catch (Exception ex)
                 {
-                    list.Add(line);
+                    throw new InvalidOperationException(string.Format("Fail to copy runtime assemblies [{0}]", ex.Message));
                 }
             }
-            
-            // First copy the required dependencies into each lib folder
-            foreach (var pair in deps)
+
+            // parse and copy lib assemblies
+            foreach (var project in sdkProjects)
             {
-                foreach (var targetClr in RUNTIME_CORECLR_TARGETS)
+                var projectFolder = Path.Combine(ROOT, "src", project);
+                var projectAssemblies = ListDependencyAssemblies(projectFolder, target.Framework, target.Framework == "aspnetcore50" ? coreclrFolder : null);
+
+                projectAssemblies.ExceptWith(runtimeAssemblies);
+
+                var targetLibDir = Path.Combine(target.TargetFolder, "lib", project);
+                if (Directory.Exists(targetLibDir))
                 {
-                    var targetLibDir = Path.Combine(targetClr, "lib", pair.Key);
-                    
-                    if (!Directory.Exists(targetLibDir)) 
+                    foreach (var file in projectAssemblies)
                     {
-                        continue;
-                    }
-                    
-                    foreach (var dep in pair.Value)
-                    {
-                        var source = Path.Combine(RUNTIME_CORECLR_WIN_x86_BIN, dep + ".dll");
-                        
-                        if (!File.Exists(source))
-                        {
-                            continue;
-                        }
-                        
-                        var dest = Path.Combine(targetLibDir, dep + ".dll");
-                        Log.Info("Copying " + source + " to " + dest);
-                        File.Copy(source, dest, true);
+                        var targetFilePath = Path.Combine(targetLibDir, Path.GetFileName(file));
+                        Log.Info("Copying " + file + " to " + targetFilePath);
+                        File.Copy(file, targetFilePath, true);
                     }
                 }
             }
-            
-            var runtimeDeps = deps["Runtime"];
-            
-            foreach(var targetClr in RUNTIME_CORECLR_TARGETS)
+
+            // copy required CoreCLR assemblies
+            if (target.Framework == "aspnetcore50")
             {
-                foreach(var file in Directory.EnumerateFiles(targetClr, "System.*.dll"))
+                foreach (var file in Directory.GetFiles(coreclrFolder).Where(f => !Path.GetFileName(f).StartsWith("System")))
```

Comment:
These are different. Originally it copies all files and then go delete `System.*.dll` if it doesn't exist in dependency list. Now it copy all files except `System.*.dll` which are copied in previous logic.